sapmon/payload/provider/prometheus.py

```diff
@@ -0,0 +1,169 @@
+# Python modules
+import json
+import logging
+
+# Payload modules
+from const import PAYLOAD_VERSION
+from helper.tools import JsonEncoder
+from provider.base import ProviderInstance, ProviderCheck
+from typing import Dict, List
+
+# provider specific modules
+from prometheus_client.samples import Sample
+from prometheus_client.parser import text_string_to_metric_families
+import uuid
+import re
+import requests
+from requests.exceptions import Timeout
+import urllib
+from datetime import datetime,timezone
+###############################################################################
+
+class prometheusProviderInstance(ProviderInstance):
+    metricsUrl = None
+    HTTP_TIMEOUT = (2, 5) # timeouts: 2s connect, 5s read
+
+    def __init__(self,
+               tracer: logging.Logger,
+               providerInstance: Dict[str, str],
+               skipContent: bool = False,
+               **kwargs):
+        super().__init__(tracer,
+                         providerInstance,
+                         skipContent,
+                         **kwargs)
+
+    def parseProperties(self):
+        ### Fixme: Should this validate the url format?
+        self.metricsUrl = self.providerProperties.get("prometheusUrl", None)
+        if not self.metricsUrl:
+            self.tracer.error("[%s] PrometheusUrl cannot be empty" % self.fullName)
+            return False
+        self.instance_name = urllib.parse.urlparse(self.metricsUrl).netloc
+        return True
+
+    def validate(self) -> bool:
+        self.tracer.info("fetching data from %s to validate connection" % self.metricsUrl)
+        return bool(self.fetch_metrics())
+
+    def fetch_metrics(self) -> str:
+        try:
+            resp = requests.get(self.metricsUrl, timeout = self.HTTP_TIMEOUT)
+            resp.raise_for_status()
+            return resp.text
+        except Exception as err:
+            self.tracer.info("Failed to fetch %s (%s)" % (self.metricsUrl, err))
+            return None
+
+    @property
+    def instance(self):
+        return self.instance_name
+
+# Implements a generic prometheus collector
+class prometheusProviderCheck(ProviderCheck):
+    colTimeGenerated = None
+    excludeRegex = re.compile(r"^(?:go|promhttp|process)_")
+    lastResult = ([], None)
+
+    def __init__(self,
+                 provider: ProviderInstance,
+                 **kwargs):
+        return super().__init__(provider, **kwargs)
+
+    def _actionFetchMetrics(self,
+                            includePrefixes: str) -> bool:
+        self.tracer.info("[%s] Fetching metrics" % self.fullName)
+        metricsData = self.providerInstance.fetch_metrics()
+        self.lastResult=(metricsData, includePrefixes)
+        if metricsData == None:
```

Comment:
As per https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none:
```suggestion
        if metricsData is None:
```